File: app/src/main/java/com/github/goldy1992/mp3player/commons/Normaliser.java
package com.github.goldy1992.mp3player.commons;

import androidx.annotation.NonNull;

import org.apache.commons.lang3.StringUtils;

public final class Normaliser {

    public static String normalise(@NonNull String query) {
        query = StringUtils.stripAccents(query);
        return query.trim().toUpperCase();
    }
}

File: commons/src/main/java/com/github/goldy1992/mp3player/commons/Normaliser.kt
package com.github.goldy1992.mp3player.commons

import org.apache.commons.lang3.StringUtils
import java.util.Locale

object Normaliser {
    @JvmStatic
    fun normalise(query: String): String {
        var toReturn = query
        toReturn = StringUtils.stripAccents(toReturn)
        return toReturn.trim { it <= ' ' }.uppercase(Locale.getDefault())
    }
}